pkg/kv/dynamodb/store.go

```diff
@@ -455,3 +463,43 @@ func (e *EntriesIterator) Err() error {
 func (e *EntriesIterator) Close() {
 	e.err = kv.ErrClosedEntries
 }
+
+// StartPeriodicCheck performs one check and continues every 'interval' in the background
+func (s *Store) StartPeriodicCheck() {
+	interval := s.params.HealthCheckInterval
+	if interval <= 0 {
+		return
+	}
+	s.ticker = time.NewTicker(interval)
+	quit := make(chan struct{})
+
+	go func() {
+		// check first and loop for checking every interval
+		logging.Default().WithField("interval", interval).Debug("Starting DynamoDB health check")
+		s.Check()
+		for {
+			select {
+			case <-s.ticker.C:
+				s.Check()
+			case <-quit:
+				return
+			}
+		}
+	}()
+}
+
+func (s *Store) Check() {
+	success, err := isTableExist(context.Background(), s.svc, s.params.TableName)
+	if success {
+		logging.Default().Debug("DynamoDB health check passed!")
+	} else {
+		logging.Default().WithError(err).Debug("DynamoDB health check failed")
+	}
+}
+
+func (s *Store) StopPeriodicCheck() {
+	if s.ticker != nil {
+		logging.Default().Info("Stopping DynamoDB health check")
+		s.ticker.Stop()
```

Comment:
1. Logging - we should create logger in the driver level which holds the name of the store type - check local adapter. When possible we should pass logger or create one relevant to the service that log information.
2. Need wait group to wait under the check goroutine is done. This is just to be nice with the caller.
3. If we signal the goroutine to stop using a channel - just close the channel and move the ticker into `StartPeriodicCheck` (not a member) + defer close, so when we close the channel in `StopPeriodcCheck` the ticker will be stopped.